### PR TITLE
Add SemanticTokenTypes.label

### LINKED ISSUE #1422: Extend predefined semantic token types to include `label`
https://github.com/microsoft/vscode-languageserver-node/blob/main/types/src/main.ts#L3805
has the list of predefined semantic token types. That matches the spec https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/

We noticed the list omits `label` which is supported by vscode.
https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#standard-token-types-and-modifiers

Our extension uses the node client library. Am I understanding it correct that there is no way for us to adjust semantic token client-side capability to customize the SemanticTokensLegend and we should make changes to the types/src/main.ts (and spec change??)

### PR DESCRIPTION
**Pull request by hyangah**
Visual Studio Code has 'label' in its standard token types list. https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#standard-token-types-and-modifiers

Fixes #1422

### CODE REVIEW

File: types/src/main.ts [hunk 1]
```diff
@@ -3832,7 +3832,12 @@ export enum SemanticTokenTypes {
 	/**
 	 * @since 3.17.0
 	 */
-	decorator = 'decorator'
+	decorator = 'decorator',
+	/**
+	 * @since 3.18.0
+	 */
+	label = 'label'
+
```
Comment (dbaeumer) [inline types/src/main.ts:3840]: Can you please remove the extra new line here.

File: client/src/common/semanticTokens.ts [hunk 2]
```diff
@@ -74,7 +74,8 @@ export class SemanticTokensFeature extends TextDocumentLanguageFeature<boolean |
     		SemanticTokenTypes.number,
     		SemanticTokenTypes.regexp,
     		SemanticTokenTypes.operator,
-			SemanticTokenTypes.decorator
+			SemanticTokenTypes.decorator,
+			SemanticTokenTypes.label,
 		];
 		capability.tokenModifiers = [
     		SemanticTokenModifiers.declaration,
```
Comment (dbaeumer) [inline client/src/common/semanticTokens.ts:81]: Can you please remove the extra semicolon here.

**Review by dbaeumer — CHANGES_REQUESTED** [review]
See comments

**Comment by hyangah** [comment]
It needs 1 more collaborator approval(s).
Is there anything else I should do to unblock this issue? Thanks!

**Pull request merged**